spec/features/allocate_feature_spec.rb

```diff
@@ -19,55 +20,64 @@
 
   let!(:case_information) {
     create(:case_information, offender: build(:offender, nomis_offender_id: nomis_offender_id), tier: 'A', case_allocation: 'NPS', probation_service: 'England')
-    create(:case_information, offender: build(:offender, nomis_offender_id: never_allocated_offender))
+    create(:case_information, offender: build(:offender, nomis_offender_id: never_allocated_offender_id))
   }
 
   before do
     signin_spo_user
   end
 
-  scenario 'accepting the recommended POM type', vcr: { cassette_name: 'prison_api/create_new_allocation_feature' } do
-    visit prison_prisoner_staff_index_path('LEI', nomis_offender_id)
+  context 'when a journey begins on the "Make new allocations" page' do
+    before do
+      visit unallocated_prison_prisoners_path('LEI')
+
+      # Takes you to the Review case and allocate page where users review case details and scroll down to select from a list of POMs
+      click_link unallocated_offender_name
+      expect(page).to have_current_path(prison_prisoner_staff_index_path('LEI', never_allocated_offender_id))
+    end
 
-    expect(page).to have_content('Determinate')
+    scenario 'accepting the recommended POM type', :js, vcr: { cassette_name: 'prison_api/create_new_allocation_feature' } do
+      expect(page).to have_content('Determinate')
 
-    # Allocate to the Probation POM called "Moic Integration-Tests"
-    within '#recommended_poms' do
-      within row_containing 'Moic Integration-Tests' do
-        click_link 'Allocate'
+      # Allocate to the Prison POM called "Moic Pom"
+      within '#recommended_poms' do
+        within row_containing 'Moic Pom' do
+          click_link 'Allocate'
+        end
       end
-    end
 
-    expect(page).to have_css('h1', text: 'Confirm allocation')
-    expect(page).to have_css('p', text: "You are allocating #{offender_name} to Moic Integration-Tests")
+      expect(page).to have_css('h1', text: 'Confirm allocation')
 
-    click_button 'Complete allocation'
+      click_button 'Complete allocation'
 
-    expect(current_url).to have_content(unallocated_prison_prisoners_path('LEI'))
-    expect(page).to have_css('.notification', text: "#{offender_name} has been allocated to Moic Integration-Tests (Probation POM)")
-  end
+      # Return to the "Make new allocation" page
+      expect(current_url).to have_content(unallocated_prison_prisoners_path('LEI'))
```

Comment:
### Possible readability improvement?

I wonder whether these tests would be more readable if we pulled the 'start page' (first page of the journey) out into a `let()` variable that we then reference at both the beginning and end of the journey.

For example, where we currently have:

```ruby
before do
  visit unallocated_prison_prisoners_path('LEI')
  # do some things...
end

it 'does the journey' do
  # do some things...
  expect(current_url).to have_content(unallocated_prison_prisoners_path('LEI'))
end
```

How about this?

```ruby
let(:start_page) { unallocated_prison_prisoners_path('LEI') }

before do
  visit start_page
  # do some things...
end

it 'does the journey' do
  # do some things...
  expect(current_url).to eq(start_page)
end
```

The main thing I like about this approach is that it should hopefully make it more obvious that we're expecting to end up back at the exact same page URL. You could say it's a bit more self-describing.

This is just my opinion, though! So completely up to you whether you choose to change it or not. (No offence taken if you don't! 🙂)

Reply:
@ollietreend Yes I like that. Definitely more readable. Thanks Ollie. :)